django_geosource/models.py

```diff
@@ -373,3 +374,53 @@ def refresh_data(self):
 
     def _get_records(self, limit=None):
         return []
+
+
+class CSVSource(Source):
+    COMA = ("coma", ",")
+    SEMI_COLON = ("semicolon", ";")
+    TAB = ("tabulation", "\t")
+    SPACE = ("space", " ")
+    COLUMN = ("column", " ")
+    QUOTATION_MARK = ("quotationmark", '"')
+    POINT = ("point", ".")
+
+    SEPARATORS = (COMA, SEMI_COLON, TAB, SPACE, COLUMN)
+    DECIMAL_SEPARATORS = (COMA, SEMI_COLON, POINT, SPACE)
+    DELIMITERS = (QUOTATION_MARK,)
+
+    file = models.FileField(upload_to="geosource/csv/%Y")
+    encoding_field = models.CharField(max_length=100)
+    decimal_separator_field = models.CharField(
+        max_length=100, choices=DECIMAL_SEPARATORS, default=POINT[1]
+    )
+    separator_field = models.CharField(
+        max_length=100, choices=SEPARATORS, default=SEMI_COLON[1]
+    )
+    delimiter_field = models.CharField(
+        max_length=100, choices=DELIMITERS, default=QUOTATION_MARK[1]
+    )
+    longitude_field = models.FloatField()
+    latitude_field = models.FloatField()
+    number_lines_to_ignore_field = models.PositiveIntegerField(default=0)
+
+    def get_file_as_sheet(self):
+        with open(self.file.name, "r") as f:
+            try:
+                return pyexcel.get_sheet(
+                    file_type="csv", file_content=f, delimiter=self.separator_field
+                )
+            except pyexcel.exceptions.FileTypeNotSupported:
+                logger.info("Source's CSV file is not valid")
```

Comment:
It's just an info log ?

Reply:
Does this information have a chance to be read by the end user ?

Reply:
The Exception is re raised with this message now.